src/nvim/edit.c

```diff
@@ -2485,6 +2485,69 @@ void set_completion(colnr_T startcol, list_T *list)
 }
 
 
+void update_completion(colnr_T startcol, list_T *list)
+{
+  // If already doing completions stop it.
```

Comment:
You mean the other way around, if completion is _not_ started yet, behave like `complete()` ?

Reply:
It just the prototype behavior to test it easily.

Reply:
I asked about the intention. That would be the _right_ thing to do, if that's the case. It just seems the comment is wrong.

Reply:
OK.  I have changed the implementation.

Reply:
@Shougo I explicitly said that that it was the _comment_ that was wrong, not the implementation. Revert the removal and change the comment to "If completion not active yet, start it" or something.

Reply:
The comment is right.  So I have changed the implementation.

Reply:
Because, `complete_update()` does not take pos argument.
It uses previous complete position.
Please see new implementation.

`complete_update()` does not replace `complete()` implementation.
It just updates the current popup.

Reply:
The old API was also better from two robustness standpoints:
- if completion was formally aborted due to some event handling racyness, it can still continue using `complete_update()`
- In a bad config two plugins might compete over the pum. In that situation `complete_update()` without pos will lead to items being interpreted at the offset, leading to incorrect items. Pum flicker by itself is of course quite bad, but flicker with garbled items is a lot worse.

What was the reason for the change?

Reply:
Hm...  I have restored the changes and fallback.